### ISSUE TITLE
HTTP Body io.Reader as parser input results in ErrUnexpectedEndOfDemo

### ISSUE THREAD

**Issue by Martig3** [markus-wa/demoinfocs-golang#299]
**Describe the bug**
Passing an HTTP Body `io.Reader` to `NewParser()` then calling `ParseToEnd()` results in an `demo stream ended unexpectedly (ErrUnexpectedEndOfDemo)` error

**To Reproduce**
I tested with https://www.dropbox.com/s/7lb1r11olhc6a1b/2021-6-6_pug_de_overpass_60bc5197ffc453a0de48212d.dem?dl=0 but I believe this is reproducible with all demos.

Code:
```go
package main

import (
	"github.com/gin-gonic/gin"
	"github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs"
)

func main() {
	r := gin.Default()
	r.POST("/parse-stats", func(c *gin.Context) {
		p := demoinfocs.NewParser(c.Request.Body)
		err := p.ParseToEnd()
		if err != nil {
			c.AbortWithStatusJSON(500, err.Error())
			return
		}
               c.JSON(200, "success")
	})
	err := r.Run()
	if err != nil {
		println(err)
		return
	} // listen and serve on 0.0.0.0:8080
}
```

**Expected behavior**
Demo gets parsed successfully as it does with `os.Open()` readers. Specifically from the code example, POSTing to `/parse-stats` should return a 200 and not panic on `p.ParseToEnd()`.

**Library version**
v2.9.0

**Comment by markus-wa** [comment]
Thanks for raising this, I will probably need to fix something in https://github.com/markus-wa/gobitread :sweat_smile:

**Comment by markus-wa** [comment]
ah, for anyone else who finds this (as discussed via gitter):
the workaround is to use [`io.ReadAll()`](https://golang.org/pkg/io/#ReadAll) for now

**Comment by markus-wa** [comment]
sorry this took so long @Martig3 , would you be able to try with this branch: https://github.com/markus-wa/demoinfocs-golang/pull/305

    go get -u github.com/markus-wa/demoinfocs-golang/v2@fix-bitread-streaming

I'm not 100% sure it will fix every case but for me I had a similar issue with gzip that went away with the change.
Would be great if you can test it with Gin as I don't have the setup (i.e. curl command or so) to then upload a demo via post at hand

**Comment by Martig3** [comment]
No problem @markus-wa, thanks for the patch. I tested this with a few of my demo files and it appears to be working in my project [branch here](https://github.com/Martig3/demo-stats-service/tree/bitread-fix-test). When I have some more time I'll write some tests on my end but it appears to be working consistently enough for me at the moment.

**Comment by markus-wa** [comment]
awesome, thanks - I'll get a release out later this week

**Comment by markus-wa** [comment]
Just released [v2.10.0](https://github.com/markus-wa/demoinfocs-golang/releases/tag/v2.10.0) which should fix it.

I'm not 100% sure if this will cover all cases as it's a bit tricky to test - but if anyone is still experiencing issues please let me know and we can re-open this :slightly_smiling_face:
